src/benchmarks/table.cpp

```diff
@@ -0,0 +1,96 @@
+#include "util/timing_util.hpp"
+
+#include "osrm/table_parameters.hpp"
+
+#include "osrm/coordinate.hpp"
+#include "osrm/engine_config.hpp"
+#include "osrm/json_container.hpp"
+#include "storage/io.hpp"
+#include "storage/serialization.hpp"
+
+#include "osrm/osrm.hpp"
+#include "osrm/status.hpp"
+
+#include <boost/assert.hpp>
+
+#include <exception>
+#include <iostream>
+#include <string>
+#include <utility>
+
+#include <cstdlib>
+
+int main(int argc, const char *argv[]) try
+{
+    if (argc < 2)
+    {
+        std::cerr << "Usage: " << argv[0] << " data.osrm\n";
+        return EXIT_FAILURE;
+    }
+
+    using namespace osrm;
+
+    std::vector<util::Coordinate> coordinates;
+    // Loading list of coordinates
+    {
+        using namespace osrm::storage;
+        io::FileReader nodes_file(std::string(argv[1]) + ".nodes",
+                                  io::FileReader::HasNoFingerprint);
+        std::size_t count = 0;
+        std::vector<OSMNodeID> nodes;
+
+        nodes_file.ReadInto(count);
+        coordinates.resize(count);
+        nodes.reserve(count);
+
+        serialization::readNodes(nodes_file, coordinates.data(), nodes, count);
+    }
+
+    // Configure based on a .osrm base path, and no datasets in shared mem from osrm-datastore
+    EngineConfig config;
+    config.storage_config = {argv[1]};
+    config.use_shared_memory = false;
+
+    // Routing machine with several services (such as Route, Table, Nearest, Trip, Match)
+    OSRM osrm{config};
+
+    // Shuffle the coordinate list randomly, then we'll simply use the first N coordinates
+    // to build our random table queries
+    std::random_shuffle(coordinates.begin(), coordinates.end());
```

Comment:
i think it must be moved to line 62 to have a random sample for every test run, at the moment it is not: page caching contains access history